Начата работа над заданием 2.

LLSnL opened the pull request:
1. Добавлено 2 класса.
2. Для классов написаны конструкторы.
3. Для класса графов написано добавление/удаление/изменение его узла.

LLSnL committed:
Начата работа над заданием 2.

1. Добавлено 2 класса.
2. Для классов написаны конструкторы.
3. Для класса графов написано добавление/удаление/изменение его узла.

LLSnL committed:
Добавлен алгоритм Ивена

LLSnL committed:
Исправлены конструкторы

LLSnL committed:
Редактирование файлов

LLSnL committed:
Добавлены темплейты

LLSnL committed:
Работа с комментариями

LLSnL committed:
Комментарии + библиотека

1. Починена работа комментариев.
2. Файлы классов перенесены в отдельную библиотеку.

LLSnL committed:
Комментарии + константы

1. Переделка функций, добавление const
2. Изменение комментариев

LLSnL committed:
Исправлена самоинкапсуляция

LLSnL committed:
Исправлена обработка ошибок

LLSnL committed:
Добавлен алгоритм Дейкстры

LLSnL committed:
Добавлены комментарии к коду

LLSnL committed:
Проектирование тестов. Начало.

Также отредактированы комментарии.

LLSnL committed:
Доделаны тесты

LLSnL committed:
Удалены неиспользуемые файлы

MarinaPaley commented on 441.2.1/441.2.1/Edge.h:
```diff
@@ -0,0 +1,80 @@
+	* @brief Сеттер size.
+	* @param newSize - новый размер ребра.
+	*/
+	void setInfo(const T newSize);
```
Странное название

MarinaPaley commented on 441.2.1/441.2.1/Edge.h:
```diff
@@ -0,0 +1,80 @@
+	* @brief Геттер size.
+	* @return Размер ребра.
+	*/
+	T getInfo() const;
```
Почему инфо?

MarinaPaley commented on 441.2.1/441.2.1/Edge.h:
```diff
@@ -0,0 +1,80 @@
+#pragma once
+#include <vector>
```
зачем здесь?

MarinaPaley commented on 441.2.1/441.2.1/Graph.h:
```diff
@@ -0,0 +1,403 @@
+	* @return true - если связность графа равна по крайней мере m,
+	* @return false - если связность меньше m.
+	*/
+	bool ivenAlgorithm(const size_t m) const;
```
Посмотрите в сторону паттерна Стратегия

LLSnL committed:
Применён паттерн Strategy

Также добавлен реализация алгоритма Клейтмана для наглядности паттерна.

LLSnL committed:
Тесты + исправления

1. Исправлены тесты.
2. Внесены исправления в классы в соответствии с результатами тестов.
3. Написан пример работы программы в Main.cpp.

MarinaPaley commented on 441.2.1/441.2.1/Graph.h:
```diff
@@ -0,0 +1,183 @@
+		if(adjacencyMatrix[i].size() != nodeCount)
+			throw std::logic_error("Размеры матрицы несовпадают с заявленным количеством узлов!");
+	}
+	double epsilon = 0.00001;
```
`std::numeric_limits<T>::epsilon`

MarinaPaley commented on 441.2.1/Library/CheckForGraphConnectivity.h:
```diff
@@ -0,0 +1,48 @@
+
+template <typename T>
+void CheckForGraphConnectivity<T>::setStrategy(CheckForGraphConnectivityStrategy<T>* s) {
+	operation = s;
```
м.б. `nullptr`

MarinaPaley commented on 441.2.1/Library/CheckForGraphConnectivity.h:
```diff
@@ -0,0 +1,48 @@
+};
+
+template <typename T>
+bool CheckForGraphConnectivity<T>::useStrategy(Graph<T> a, size_t nodeNumber) {
```
Вам буковок не отсыпали? Пишите, пожалуйста, имена полностью.

![image](https://user-images.githubusercontent.com/26978698/143503209-23fadee5-c3a0-4cb2-9eb8-dc2e1bdffa79.png)

MarinaPaley commented on 441.2.1/Library/IvenAlgorithm.h:
```diff
@@ -0,0 +1,194 @@
+* @brief Класс - конкретная реализация алгоритма проверки графа на связность - реализация алгоритма Ивена.
+*/
+template <typename T>
+class IvenAlgorithm : public CheckForGraphConnectivityStrategy<T> {
```
`final`

MarinaPaley commented on 441.2.1/Library/KleitmanAlgorithm.h:
```diff
@@ -0,0 +1,179 @@
+	* @return true - если маршрут между узлами существует,
+	* @return false - если нет.
+	*/
+	static bool BFS(Graph<T> g, size_t to, std::vector<bool>& visited, std::list<size_t>& buffer, std::vector<size_t>& parents);
```
Просто подумайте об архитектуре. `static` плохо

LLSnL committed:
Внесены исправления

LLSnL committed:
Внесены исправления
